autopush/router/gcm.py

```diff
@@ -43,12 +43,14 @@ def amend_msg(self, msg):
         msg["senderid"] = self.creds.get("senderID")
         return msg
 
-    def register(self, uaid, router_data):
-        """Validate that a token is in the ``router_data``"""
-        if not router_data.get("token"):
-            raise self._error("connect info missing 'token'", status=401)
+    def register(self, uaid, router_data, router_token=None, *kwargs):
+        """ Validate that the GCM Instance Token is in the ``router_data``"""
+        if "token" not in router_data:
+            raise self._error("connect info missing GCM Instance 'token'",
+                              status=401)
         # Assign a senderid
-        router_data["creds"] = self.creds = self.senderIDs.choose_ID()
+        router_data["creds"] = self.creds = \
```

Comment:
``self.creds``, which is on a single router instance used concurrently from multiple paths keeps changing its state based on the router_token of the last register call?